server/lib/matchmaking/matchmaking-preferences-api.ts

```diff
@@ -39,6 +39,10 @@ export class MatchmakingPreferencesApi {
       throw new httpErrors.BadRequest('invalid matchmaking type')
     }
 
+    if (body.mapSelections && body.mapSelections.length > currentMapPool.maxVetoCount) {
```

Comment:
This error is also bad here. These things made sense when the veto count was static but since it can change between map pools and matchmaking preferences are saved across pools, this is a problem